### ISSUE TITLE
XSRF-TOKEN cookie does not include appname or path

### ISSUE THREAD

**Issue by matthijskooijman** [LycheeOrg/Lychee-Laravel#325]
When two Lychee installations live in different subdirectories of the same domain, their XSRF-TOKEN cookies conflict (overwrite each other) and cause problems.

I'm not sure if it is actually a supported configuration to run Lychee in a subdir, but other than this issue, it has been running fine for me so far.

The observable problem is that when you open one instance, then another in a second tab, then navigate the first one to "Settings" and then "More", you get a 500 error on the Settings::getAll POST request.

I noticed that this issue does not occur for the login status, since that seems to be put into the session and the session cookie uses the APP_NAME to allow distinguishing multiple installations (it might still be better to use a path-based cookies, but that might require more info on how Lychee is installed and might get complicated when internal rewrites are involved). I wonder: Wouldn't it make sense to put the XSRF-token inside the session anyway?

As a sidenote: It took me a while to figure out what was wrong, since this situation did not produce any log message at all (not in the PHP log, nor in the Laravel log).

**Comment by ildyria** [comment]
I guess here is the script to be fixed: https://github.com/LycheeOrg/Lychee-front/blob/master/scripts/csrf_protection.js

**Comment by kamil4** [comment]
Yeah, I noticed that behavior (I have my production installation on the same server as the development ones) but I assumed it was an inherent limitation and learnt to live with it. Thanks for the hint about `APP_NAME` -- that will go a long way for me! If you figure out a foolproof solution, do not hesitate to submit a PR! For instance, would using `APP_URL` instead of `APP_NAME` work?
